spec/spec_helper.rb

```diff
@@ -23,9 +24,26 @@ def stub_api_request method, uri, fixture = nil
     end
     stub_request(method, uri.to_s).to_return response
   end
+  
+  def environment
+    Recurly.subdomain = 'api'
+    Recurly.api_key = 'api_key'
+    Recurly.default_currency = 'USD'
+    Recurly.logger = Logger.new nil
+  end
 end
+
 include SpecHelper
 
+environment
+
+class Minitest::Spec
+  after do |tests|
+    WebMock.reset!
+    environment
```

Comment:
Looks like line 38 can be killed and I will do so. Reading your question, the method call clearly is not expressive enough. It should be called ```reset_recurly_environment```. The purpose of the call is initialize the environment to a know state. The source of the need for this was two fold. In my local environment the tests were a bit flakey in that some times they would pass and others they would not. I added the after block to do two things:

1. Cleanup WebMocks - Each time a request was being stubbed, it would bleed over into other tests. I.e. Mocks created in one test were available in other tests. This behavior creates a lack of clarity and in this case some flaky tests. 

2. Environment - The recurly gem uses global state to store things like API tokens, domains, etc. There were some test that were resetting the global state in order to test the behavior of the global state. As a matter or practice, I thought that it was important to make sure that each test (even if it is not twiddling global state) reset the global state to a know good value after the completion of the test. Doing so eliminates an entire set of spec bugs that are slightly difficult to diagnose and ad in this case will cause tests to fail depending on the order that they are executed.